### ISSUE TITLE
Errors when opening a file with a `gql` tag; no syntax highlighting

### ISSUE THREAD

**Issue by mgeraci** [jparise/vim-graphql#35]
Hi there! I recently updated a `vim-graphql`, and started getting errors when opening files with `gql` tags, and no syntax highlighting is shown. The errors are:

```
Error detected while processing /home/u0/mgeraci/.vim/bundle/vim-graphql/after/syntax/javascript/graphql.vim:
line   10:
E117: Unknown function: graphql#javascript_tags
E116: Invalid arguments for function join(graphql#javascript_tags(), '\|') . '\)'
E15: Invalid expression: '\%(' . join(graphql#javascript_tags(), '\|') . '\)'
line   12:
E121: Undefined variable: s:tags
E15: Invalid expression: 'syntax region graphqlTemplateString start=+' . s:tags . '\@20<=`+ skip=+\\`+ end=+`+ contains=@GraphQLSyntax,jsTemplateExpression,jsSpecial extend'
line   13:
E121: Undefined variable: s:tags
E15: Invalid expression: 'syntax match graphqlTaggedTemplate +' . s:tags . '\ze`+ nextgroup=graphqlTemplateString'
line   10:
E117: Unknown function: graphql#javascript_tags
E116: Invalid arguments for function join(graphql#javascript_tags(), '\|') . '\)'
E15: Invalid expression: '\%(' . join(graphql#javascript_tags(), '\|') . '\)'
line   12:
E121: Undefined variable: s:tags
E15: Invalid expression: 'syntax region graphqlTemplateString start=+' . s:tags . '\@20<=`+ skip=+\\`+ end=+`+ contains=@GraphQLSyntax,jsTemplateExpression,jsSpecial extend'
line   13:
E121: Undefined variable: s:tags
E15: Invalid expression: 'syntax match graphqlTaggedTemplate +' . s:tags . '\ze`+ nextgroup=graphqlTemplateString'
Press ENTER or type command to continue
```

I tried checking out a few different commits of the plugin, and found that the breaking change was in https://github.com/jparise/vim-graphql/commit/1ad3aa6f9fad65c646fdabe057ea883c34a73a3d (the last working version is `792c7bcb138c1e787a7527f16ce63e8cf53df6ba`). I'm seeing this error with any commit after the working version, including at master.

It's possible that there's some odd interaction with other plugins; let me know what I can do to help debug this issue!

### Vim Version

```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jun 21 2019 04:10:35)
Included patches: 1-197, 322, 377-378, 550, 649, 651, 703, 706-707
Extra patches: 8.1.1401, 8.1.1382, 8.1.1368, 8.1.1367, 8.1.1366, 8.1.1365, 8.1.1046, 8.1.0613, 8.1.0547, 8.1.0546, 8.1.0544, 8.1.0540, 8.1.0539, 8.1.0538, 8.1.0506, 8.1.0208, 8.1.0206, 8.1.0205, 8.1.0189, 8.1.0177, 8.1.0067, 8.1.0066
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by pkg-vim-maintainers@lists.alioth.debian.org
Huge version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     +termguicolors
-browse          -footer          +multi_byte      +terminfo
++builtin_terms  +fork()          +multi_lang      +termresponse
+byte_offset     +gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
-clientserver    +insert_expand   +packages        -toolbar
-clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        -perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       -python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             +rightleft       +writebackup
+diff            +menu            -ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xpm
+emacs_tags      -mouseshape      +startuptime     -xsmp
+eval            +mouse_dec       +statusline      -xterm_clipboard
+ex_extra        +mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax
+farsi           +mouse_netterm   +tag_binary
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -Wdate-time  -g -O2 -fdebug-prefix-map=/build/vim-xBMBkh/vim-8.0.0197=. -fstack-protector-strong -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -lselinux -lacl -lattr -lgpm -ldl
```

**Comment by jparise** [comment]
@mgeraci what does `:echo graphql#javascript_tags()` (on your vim command line) output?

**Comment by mgeraci** [comment]
From the [working commit](https://github.com/jparise/vim-graphql/commit/792c7bcb138c1e787a7527f16ce63e8cf53df6ba):
```
E117: Unknown function: graphql#javascript_tags
```

And from the [broken commit](https://github.com/jparise/vim-graphql/commit/1ad3aa6f9fad65c646fdabe057ea883c34a73a3d):
```
E117: Unknown function: graphql#javascript_tags
```

So it seems like regardless of the `vim-graphql` version, I'm missing that function.

**Comment by jparise** [comment]
1ad3aa6f9fad65c646fdabe057ea883c34a73a3d introduced the `autoload/graphql.vim` file, which is where that function is defined. If that file hasn't made it into your `runtimepath`, that would explain the problem you're seeing.

Do you have that file in your `.vim/bundle/` path (as `/home/u0/mgeraci/.vim/bundle/vim-graphql/autoload/graphql.vim`)? Are you using a plugin manager or some other tool to update your plugins?

**Comment by mgeraci** [comment]
When I check out `1ad3aa6f9fad65c646fdabe057ea883c34a73a3d`, I do see that file created in my vim bundle folder. So that would definitely imply that there is something wrong in my tooling!

I'm using pathogen to load vim plugins, and [homeshick](https://github.com/andsens/homeshick) to sync and symlink dotfiles (including vim plugins as submodules in my dotfile repo).

And thinking about that actually led me to the resolution of the issue... Because symlinks aren't dynamic or anything, when I updated your plugin and it added a new file, I needed to re-link my dotfiles.

So, in closing, there is nothing at all wrong with your code, it was all on me! But I really appreciate you taking the time to respond and help me debug this. Thank you!
